### ISSUE TITLE
fix: template uses lang html for template, however it is not installed

### ISSUE THREAD

**Issue by dargmuesli** [nuxt/framework#7256]
### Environment

- Operating System: `Linux`
- Node Version:     `v14.19.3`
- Nuxt Version:     `3.0.0-rc.9`
- Nitro Version:    `0.5.1`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `telemetry`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://codesandbox.io/s/template-lang-html-dzr7f2?file=/pages/index.vue

### Describe the bug

Setting `lang="html"` on a template tag yields the following error in `@vitejs/plugin-vue`:
```
 ERROR  Component /.../pages/index.vue uses lang html for template, however it is not installed.         19:16:56

  at error (node_modules/.pnpm/rollup@2.77.3/node_modules/rollup/dist/es/shared/rollup.js:1858:30)
  at throwPluginError (node_modules/.pnpm/rollup@2.77.3/node_modules/rollup/dist/es/shared/rollup.js:21764:12)
  at Object.error (node_modules/.pnpm/rollup@2.77.3/node_modules/rollup/dist/es/shared/rollup.js:22486:20)
  at Object.error (node_modules/.pnpm/rollup@2.77.3/node_modules/rollup/dist/es/shared/rollup.js:21941:42)
  at node_modules/.pnpm/@vitejs+plugin-vue@3.1.0_vite@3.0.9+vue@3.2.38/node_modules/@vitejs/plugin-vue/dist/index.mjs:161:32
  at Array.forEach (<anonymous>)
  at compile (node_modules/.pnpm/@vitejs+plugin-vue@3.1.0_vite@3.0.9+vue@3.2.38/node_modules/@vitejs/plugin-vue/dist/index.mjs:160:19)
  at transformTemplateAsModule (node_modules/.pnpm/@vitejs+plugin-vue@3.1.0_vite@3.0.9+vue@3.2.38/node_modules/@vitejs/plugin-vue/dist/index.mjs:130:18)
  at Object.transform (node_modules/.pnpm/@vitejs+plugin-vue@3.1.0_vite@3.0.9+vue@3.2.38/node_modules/@vitejs/plugin-vue/dist/index.mjs:2653:18)
  at node_modules/.pnpm/rollup@2.77.3/node_modules/rollup/dist/es/shared/rollup.js:22695:37
```

### Additional context

I'm not sure if that's an upstream issue. What do you think?

There was a similar issue, which was closed with a workaround: https://github.com/vuejs/rollup-plugin-vue/issues/256

**Comment by manniL** [comment]
What is the reason to use `lang="html"` in the first place? ☺️

**Comment by dargmuesli** [comment]
First of all, that's a fair question on one hand, but a little off-topic to me on the other. Setting the default shouldn't make the build crash, I'd say.

To answer your question though, [there are libraries that set the `lang` like this](https://github.com/gruhn/vue-qrcode-reader/blob/ea2cdaa785253368684d7d823b3b0416b2d5461e/src/components/QrcodeCapture.vue#L1), blocking a migration to nuxt 3. Those libraries are not necessarily regularly maintained, making the workaround of removing the tag difficult.
Also I could imagine people writing their templates in `pug` code, setting the default to `vue` and wanting some files to be kept in `html`. Maybe time for a rewrite is missing or such.

In summary I come back to my first point:
> It walks like a bug and it quacks like a bug, then it must be a bug.

**Comment by danielroe** [comment]
This _may_ be a bug, but it's not a Nuxt one. See [this reproduction](https://stackblitz.com/edit/vitejs-vite-vcvrbu?file=index.html,src%2FApp.vue&terminal=dev) with pure vite+vue. To be honest, I'm not sure whether `lang="html"` is supported in Vue 3. Would you check the issues and raise it [upstream](https://github.com/vitejs/vite)?

The library you are mentioning is a Vue 2 library, FYI.

**Comment by dargmuesli** [comment]
:heavy_check_mark:  Did so: https://github.com/vitejs/vite/issues/10033

Vue 3 docs specify such preprocessors under their [SFC Syntax](https://vuejs.org/api/sfc-spec.html#pre-processors).

The last given reproduction does not have a `lang` attribute set for me, is it the correct link? I created this for now: https://stackblitz.com/edit/vitejs-vite-wdafsc?file=src/App.vue

The library I linked does support Vue 3 under https://github.com/gruhn/vue-qrcode-reader/tree/vue3-compatibility / https://www.npmjs.com/package/vue-qrcode-reader/v/3.1.0-vue3-compatibility.2

**Comment by Jaklik** [comment]
try this one, works fine with Vue 3
https://www.npmjs.com/package/qrcode-vue3

**Comment by dargmuesli** [comment]
Well, this one says
> "JavaScript library for **generating** QR codes"

and the one I'm using
> "Vue.js components, allowing you to **detect and decode** QR codes"

so I'd stay with my choice for now.
